### PR TITLE
RC jobs for Jenkins

### PR DESCRIPTION
**Pull request by manasagowri**
# Description
Create RC pipeline scripts for Jenkins jobs execution.
Success Builds : 
1. https://ceph-downstream-jenkins-csb-storage.apps.ocp4.prod.psi.redhat.com/view/RHCS%20RC/job/rhceph-rc-rpm/3/console
2. https://ceph-downstream-jenkins-csb-storage.apps.ocp4.prod.psi.redhat.com/view/RHCS%20RC/job/rhceph-rc-rpm/6/console
3. https://ceph-downstream-jenkins-csb-storage.apps.ocp4.prod.psi.redhat.com/view/RHCS%20RC/job/rhceph-rc-container/2/console
4. 

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

File: pipeline/vars/common.groovy [hunk 1]
```diff
@@ -435,4 +441,98 @@ def getComposeId(def osVersion) {
...
+        println "Missing RC Json File."
+        return
+    }
+    def repoDetails = composeMap.build.extra.image
+    def containerImage = repoDetails.index.pull.find({x -> !(x.contains("sha"))})
+    def repoUrl = repoDetails.yum_repourls.find({x -> x.contains("RHCEPH")})
+    def composeId = repoUrl.split("/").find({x -> x.contains("RHCEPH-${rhcsVersion}")})
+    def compose_ids = rcJsonDetails.rcMapExisting.compose_urls.collect{it.compose_id}
+    if(!(composeId in compose_ids)){
+        println "Compose ID mismatch"
```
Comment (psathyan) [inline pipeline/vars/common.groovy:504]: It is better to throw an error as we don't have the complete information.

**Review by psathyan — COMMENTED** [review]
Nice. There are some minor improvements that we could done...

File: pipeline/RC/Jenkinsfile-RC-container.groovy [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                if (!jsonContent) {
+                    sh "exit 1"
+                }
+            }
+        }
+    }
+
+    stage("Publish Results"){
+        script{
+            if(jsonContent){
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/RC/Jenkinsfile-RC-container.groovy:51]: No, I am just failing the previous stage, i have handled the error with catchError. I could change that to failing the stage and remove this condition here.

File: pipeline/RC/Jenkinsfile-RC-container.groovy [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    jobName = 'rhceph-4-tier-0'
+                }else{
+                    jobName = 'rhceph-5-tier-0'
+                }
+                build wait: false, job: jobName, parameters: [string(name: 'CI_MESSAGE', value: jsonContent)]
+            }
+        }
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/RC/Jenkinsfile-RC-container.groovy:67]: done

File: pipeline/RC/Jenkinsfile-RC-container.groovy [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                def composeMap = readJSON text: "${jsonContent}"
+                def composeId = composeMap.compose_id.toString()
+                def composeUrl = composeMap.compose_url.toString()
+                def repository = composeMap.repository.toString()
+                def jobName = ""
+                if(composeMap.compose_id.contains('4')){
+                    jobName = 'rhceph-4-tier-0'
+                }else{
+                    jobName = 'rhceph-5-tier-0'
+                }
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/RC/Jenkinsfile-RC-container.groovy:61]: done

File: pipeline/RC/Jenkinsfile-RC-rpm.groovy [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    stage('Publish RC RPM Compose') {
+        script {
+            echo "${params.CI_MESSAGE}"
+            sharedLib.postRCCompose("rpm")
+        }
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/RC/Jenkinsfile-RC-rpm.groovy:44]: done

File: pipeline/vars/common.groovy [hunk 6]
```diff
@@ -419,8 +419,14 @@ def getBaseUrl(def osVersion) {
         Return the compose url for the current RHCS build. The osVersion determines the
         platform for which the URL needs to be retrieved.
     */
+    def url
+    def composeMap = readJSON text: "${params.CI_MESSAGE}"
+    if(composeMap.payload){
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/vars/common.groovy:424]: done

File: pipeline/vars/common.groovy [hunk 7]
```diff
@@ -435,4 +441,98 @@ def getComposeId(def osVersion) {
...
+    echo "in postRCRPMCompose"
+    def composeMap = getCIMessageMap()
+    def rhcsVersion = composeMap["compose-id"].substring(7,10).toLowerCase()
+    def rcJsonDetails = fetchRCJsonFile(rhcsVersion)
+    def osVersion = composeMap["compose-id"].substring(11,17)
+    def compose_path = composeMap["compose-path"].replace("/mnt/redhat/", "")
+    def compose_url = "http://download-node-02.eng.bos.redhat.com/${compose_path}".toString()
+
+    def jsonContent
+    if(rcJsonDetails.fileExists == 0 && composeMap["compose-label"] in rcJsonDetails.rcMapExisting.compose_label){
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/vars/common.groovy:467]: done

File: pipeline/vars/common.groovy (hunk 1 above)
Comment (manasagowri) [reply pipeline/vars/common.groovy:504]: done

File: pipeline/vars/common.groovy [hunk 8]
```diff
@@ -435,4 +441,98 @@ def getComposeId(def osVersion) {
...
+    jsonContent = writeJSON returnText: true, json: rcMap
+    postCompose(jsonContent.trim(), rcJsonDetails.fileName)
+}
+
+def postRCContCompose(){
+    def composeMap = readJSON text: "${params.CI_MESSAGE}"
+    def rhcsVersion = composeMap.tag.name.substring(5,8)
+    def rcJsonDetails = fetchRCJsonFile(rhcsVersion)
+    if(rcJsonDetails.fileExists != 0){
+        println "Missing RC Json File."
```
[the remark this answers is not in the document]
Comment (manasagowri) [reply pipeline/vars/common.groovy:495]: done

**Review by psathyan — APPROVED** [review]

**Pull request merged**